alica_engine/src/engine/Assignment.cpp

```diff
@@ -53,33 +56,19 @@ Assignment::Assignment(PartialAssignment* pa) {
         this->epRobotsMapping = new AssignmentCollection(assCol->getSize());
     }
 
-    shared_ptr<vector<const supplementary::AgentID*>> curRobots;
     for (short i = 0; i < this->epRobotsMapping->getSize(); i++) {
         // set the entrypoint
-        if (!this->epRobotsMapping->setEp(i, assCol->getEp(i))) {
-            cerr << "Ass: AssignmentCollection Index out of entrypoints bounds!" << endl;
-            throw new exception();
-        }
-
+        epRobotsMapping->setEp(i, assCol->getEp(i));
         // copy robots
-        auto robots = assCol->getRobots(i);
-        curRobots = make_shared<vector<const supplementary::AgentID*>>();
-        for (auto& rob : *robots) {
-            curRobots->push_back(rob);
-        }
-
+        *(epRobotsMapping->editRobots(i)) = *(assCol->getRobots(i));
         // set the robots
```

Comment:
I assume setRobots is called in editRobots?

Reply:
removed superfluous comment